test/controllers/user_controller_test.exs

```diff
@@ -115,4 +115,42 @@ defmodule CodeCorps.UserControllerTest do
     assert response(conn, 204)
     refute Repo.get(User, user.id)
   end
+
+  test "shows email availability and validity", %{conn: conn} do
+    resp = get conn, user_path(conn, :email_available, %{email: "available@mail.com"})
+
+    json = json_response(resp, 200)
+    assert json["available"]
+    assert json["valid"]
+
+    insert_user(%{email: "used@mail.com"})
+    resp = get conn, user_path(conn, :email_available, %{email: "used@mail.com"})
+    json = json_response(resp, 200)
+    refute json["available"]
+    assert json["valid"]
+
+    resp = get conn, user_path(conn, :email_available, %{email: "not_an_email"})
+    json = json_response(resp, 200)
+    assert json["available"]
+    refute json["valid"]
+  end
+
+  test "shows username availability and validity", %{conn: conn} do
+    resp = get conn, user_path(conn, :username_available, %{username: "available"})
+
+    json = json_response(resp, 200)
+    assert json["available"]
+    assert json["valid"]
+
+    insert_user(%{username: "used"})
+    resp = get conn, user_path(conn, :username_available, %{username: "used"})
+    json = json_response(resp, 200)
+    refute json["available"]
+    assert json["valid"]
+
+    resp = get conn, user_path(conn, :username_available, %{username: ""})
+    json = json_response(resp, 200)
+    assert json["available"]
+    refute json["valid"]
+  end
```

Comment:
Following our conversation earlier, I think I would wrap the current tests as `describe` blocks for the endpoint and then separate each of the three tests (valid/available, invalid/available, valid/unavailable) into their own `test` blocks.